jni_gen/lib/src/bindings/c_bindings.dart

```diff
@@ -0,0 +1,362 @@
+    }
+
+    final callType = _typeNameAtCallSite(m.returnType);
+    final callArgs = _callArgs(m, classVar, methodVar);
+    if (isACtor) {
+      s.write('(*jniEnv)->NewObject($callArgs);\n');
+    } else {
+      final ifStatic = isStatic ? 'Static' : '';
+      s.write('(*jniEnv)->Call$ifStatic${callType}Method($callArgs);\n');
+    }
+    s.write(_destroyParams(m));
+    if (returnTypeName != 'void') {
+      s.write(_result(m));
+    }
+    s.write('}\n');
+    return s.toString();
+  }
+
+  String _field(ClassDecl c, Field f) {
+    final cClassName = mangledClassName(c);
+    final isStatic = isStaticField(f);
+
+    // if the field is final and default is assigned, then no need to wrap
+    // this field. It should then be a constant in dart code.
+    if (isStatic && isFinalField(f) && f.defaultValue != null) {
+      return "";
+    }
+
+    final s = StringBuffer();
+
+    final fieldName = f.finalName;
+    final fieldVar = "${_fieldVarPrefix}_${cClassName}_$fieldName";
+
+    s.write('jfieldID $fieldVar = NULL;\n');
+    final classVar = '${_classVarPrefix}_$cClassName';
+
+    void writeAccessor({bool isSetter = false}) {
+      final ct = isSetter ? 'void' : cType(f.type.name);
+      // Getter
+      final prefix = isSetter ? 'set' : 'get';
+      s.write('$ct ${prefix}_${memberNameInC(c, fieldName)}(');
+      final formalArgs = <String>[
+        if (!isStatic) 'jobject self_',
+        if (isSetter) '${cType(f.type.name)} value',
+      ];
+      s.write(formalArgs.join(', '));
+      s.write(') {\n');
+      s.write(_loadEnvCall);
+      s.write(_loadClassCall(classVar, _internalName(c.binaryName)));
+
+      var ifStatic = isStatic ? 'static_' : '';
+      s.write(
+          '${_indent}load_${ifStatic}field($classVar, &$fieldVar, "$fieldName",'
+          '"${_fieldSignature(f)}");\n');
+
+      ifStatic = isStatic ? 'Static' : '';
+      final callType = _typeNameAtCallSite(f.type);
+      final acc = isSetter ? 'Set' : 'Get';
+      final ret = isSetter ? '' : 'return ';
+      final conv = !isSetter && !isPrimitive(f.type) ? 'to_global_ref' : '';
+      s.write('$_indent$ret$conv((*jniEnv)->$acc$ifStatic${callType}Field');
+      final secondArg = isStatic ? classVar : 'self_';
+      s.write('(jniEnv, $secondArg, $fieldVar');
+      if (isSetter) {
+        s.write(', value');
+      }
+      s.write('));\n');
+      // TODO(#25): Check Exceptions
+      s.write('}\n\n');
+    }
+
+    writeAccessor(isSetter: false);
+    if (isFinalField(f)) {
+      return s.toString();
+    }
+    writeAccessor(isSetter: true);
+    return s.toString();
+  }
+
+  final String _loadEnvCall = '${_indent}load_env();\n';
+
+  String _loadClassCall(String classVar, String internalName) {
+    return '${_indent}load_class_gr(&$classVar, '
+        '"$internalName");\n';
+  }
+
+  String _formalArgs(Method m) {
+    final args = <String>[];
+    // if normal instance method, jobject self
```

Comment:
Removed comment.

In case of constructor, there's no self param but it's actually an instance method. (Impl detail: see https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#NewObject) So hasSelfParam is separate from static / instance check.